openedx/adg/lms/webinars/admin.py

```diff
@@ -2,35 +2,86 @@
 Registering models for webinars app.
 """
 from django.contrib import admin
+from django.db.models import Q
+from django.utils.translation import ugettext_lazy as _
 
 from openedx.adg.lms.applications.admin import adg_admin_site
 
-from .models import Webinar, WebinarRegistration
+from .models import CancelledWebinar, Webinar, WebinarRegistration
 
 
-@admin.register(Webinar)
-class WebinarAdmin(admin.ModelAdmin):
+class ActiveWebinarStatusFilter(admin.SimpleListFilter):
     """
-    Model admin for webinar
+    Custom filter to provide `Upcoming` and `Delivered` states filter functionality to the WebinarAdmin
     """
 
-    save_as = True
+    title = _('Webinar Status')
+    parameter_name = 'status'
+
+    def lookups(self, request, model_admin):
+        return (
+            (Webinar.UPCOMING, _('Upcoming')),
+            (Webinar.DELIVERED, _('Delivered')),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value() == Webinar.UPCOMING:
+            return queryset.filter(status=Webinar.UPCOMING)
+
+        if self.value() == Webinar.DELIVERED:
+            return queryset.filter(status=Webinar.DELIVERED)
+
+
+class WebinarAdminBase(admin.ModelAdmin):
+    """
+    Base Model admin for webinars i.e Cancelled Webinars and Non-Cancelled Webinars
+    """
 
     list_display = ('title', 'start_time', 'presenter', 'status',)
-    list_filter = ('start_time', 'status', 'language',)
     search_fields = ('title',)
-    readonly_fields = ('created_by', 'modified_by',)
     filter_horizontal = ('co_hosts', 'panelists',)
 
+
+class WebinarAdmin(WebinarAdminBase):
+    """
+    Admin for upcoming and delivered webinars
+    """
+
+    save_as = True
+    list_filter = ('start_time', 'language', ActiveWebinarStatusFilter)
+    readonly_fields = ('created_by', 'modified_by', 'status',)
+
     def save_model(self, request, obj, form, change):
         if not change:
             obj.created_by = request.user
         else:
             obj.modified_by = request.user
         obj.save()
 
+    def get_queryset(self, request):
+        qs = super(WebinarAdmin, self).get_queryset(request)
+        return qs.filter(~Q(status=Webinar.CANCELLED))
+
+
+class CancelledWebinarAdmin(WebinarAdminBase):
+    """
+    Model admin for cancelled webinar
+    """
+
+    def get_queryset(self, request):
```

Comment:
Do we need to allow cloning of canceled events? If so we need `save_as = True`, or we can move it to base class

Reply:
No, we decided not to allow anything with cancelled events. It will simply have view permissions.